blocks/commerce-b2b-po-approval-rule-details/commerce-b2b-po-approval-rule-details.js

```diff
@@ -0,0 +1,71 @@
+import { render as purchaseOrderRenderer } from '@dropins/storefront-purchase-order/render.js';
+import { ApprovalRuleDetails } from '@dropins/storefront-purchase-order/containers/ApprovalRuleDetails.js';
+import { events } from '@dropins/tools/event-bus.js';
+import {
+  checkIsAuthenticated,
+  CUSTOMER_LOGIN_PATH,
+  CUSTOMER_PATH,
+  CUSTOMER_PO_RULES_PATH,
+  PO_PERMISSIONS,
+  rootLink,
+} from '../../scripts/commerce.js';
+
+// Initialize
+import '../../scripts/initializers/purchase-order.js';
+
+const redirectToLogin = () => {
+  window.location.href = rootLink(CUSTOMER_LOGIN_PATH);
+};
+
+const redirectToAccountDashboard = () => {
+  window.location.href = rootLink(CUSTOMER_PATH);
+};
+
+const redirectToApprovalRulesList = () => {
+  window.location.href = rootLink(CUSTOMER_PO_RULES_PATH);
+};
+
+/**
+ * Initializes and decorates the Approval Rule Details block
+ * Redirects unauthenticated users and handles permission updates
+ */
+const renderApprovalRuleDetails = async (
+  blockElement,
+  permissions = {},
+) => {
+  const hasAccess = permissions.admin || permissions[PO_PERMISSIONS.VIEW_RULES];
+  if (!hasAccess) {
+    redirectToAccountDashboard();
+  }
+
+  const approvalRuleID = new URLSearchParams(window.location.search).get('ruleRef') || '';
+  if (!approvalRuleID) {
+    redirectToApprovalRulesList();
+  }
+
+  await purchaseOrderRenderer.render(ApprovalRuleDetails, {
+    approvalRuleID,
+    routeApprovalRulesList: () => rootLink(CUSTOMER_PO_RULES_PATH),
+  })(blockElement);
+};
+
+export default async function decorate(block) {
+  // Redirect guest users
+  if (!checkIsAuthenticated()) {
+    redirectToLogin();
```

Comment:
missing return, otherwise  the code continues executing
```suggestion
    redirectToLogin();
    return;
```

Reply:
Fixed